### ISSUE TITLE
Improvements around for the PosixPermissions api

### ISSUE THREAD

**Issue by daddykotex** [typelevel/fs2#3388]
The other day I was implementing some function that did the following:

1. download some binary into a cache (via coursier)
2. copy the binary to a target path
3. set the executable bits on copied file

The only thing that was awkward to implement was 3). The shortest snippet I found to do the equiavalent of `chmod +x file` via fs2-io was:

```scala
//> using lib "co.fs2::fs2-io:3.5.0"
import cats.effect.*
import fs2.io.file.*

object Main extends IOApp.Simple {
  val run = {
    val path = Path("file")
    val getPerms = Files[IO].getPosixPermissions(path)
    getPerms flatMap { perm =>
      val permX = PosixPermissions
        .fromInt(
          perm.value |
            PosixPermission.OwnerExecute.value |
            PosixPermission.GroupExecute.value |
            PosixPermission.OthersExecute.value
        )
        .getOrElse(perm)
      Files[IO].setPosixPermissions(path, permX)
    }
  }
}
```

Add that point, I was wondering why there is not something like:

```scala
//> using lib "co.fs2::fs2-io:3.5.0"
import cats.effect.*
import fs2.io.file.*

object Main extends IOApp.Simple {
  val run = {
    val path = Path("file")
    val getPerms = Files[IO].getPosixPermissions(path)
    getPerms flatMap { perm =>
      val permX = perm
        .add(PosixPermission.GroupExecute)
        .add(PosixPermission.OwnerExecute)
        .add(PosixPermission.OthersExecute)
      Files[IO].setPosixPermissions(path, permX)
    }
  }
}
```

Is it because of something technical I would not be aware? If not could I work on some changes to the API to help with such a use case? Possibly also allow the use case of _remove_ a posix permission?

Anyway, thanks for the project, let me know if I can help

**Comment by mpilquist** [comment]
Adding some builder methods to `PosixPermissions` sounds good to me. Both `add` and `remove` maybe?

I could also see something like: `p.modify("+x")` or `PosixPermissionChange.fromString("+x").traverse(p.modify)` .

**Comment by mpilquist** [comment]
Fixed by #3396

**Comment by daddykotex** [comment]
I just spent some time working on the additional changes we talked about. and TIL, that chmod supports a lot more than I had anticipated, [see the docs](https://linux.die.net/man/1/chmod).

For example one can do: `u=g` to set the [u]sers permissions to whatever the [g]roup permissions are. You can also do `+x,+w`.

Do we want to match what chmod can accepts, or allowing a subset of that would be reasonable? eg:
1. "+x" add executable bit to all
2. "u-r" removes the read bit for users
3. "g=rw" sets the group permission as "rw-" (remove executable if it's there)
